### PR TITLE
Enabling HTTP Basic Auth

### PR DESCRIPTION
**Pull request by adamgoose**
- Enabling HTTP Basic Auth
- Tweaking argument validation
- Setting reasonable defaults
- Rebasing now

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,6 +106,9 @@ Below is the complete list of available options that can be used to customize yo
 - **TORAN_TRACK_DOWNLOADS**: Track private package installs, set to `true` to get an install log in `/data/toran-proxy/logs/downloads.private.log`. Defaults to `false`
 - **TORAN_MONO_REPO**: Set to `true` for switch to a monorepo instead of dual repo model. Defaults to `false`
 - **PHP_TIMEZONE**: Configure timezone PHP. Default `Europe/Paris`.
+- **TORAN_AUTH_ENABLE**: Set to `true` to enable HTTP Basic Authentication. When enabled, `TORAN_AUTH_USER` and `TORAN_AUTH_PASSWORD` are required.
```
Comment (cedvan) [inline README.md]: miss : Default to `false`.

File: scripts/config.sh [hunk 2]
```diff
@@ -23,6 +23,9 @@ TORAN_TOKEN_GITHUB=${TORAN_TOKEN_GITHUB:-false}
 TORAN_TRACK_DOWNLOADS=${TORAN_TRACK_DOWNLOADS:-false}
 TORAN_MONO_REPO=${TORAN_MONO_REPO:-false}
 TORAN_SECRET=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)
+TORAN_AUTH_ENABLE=${TORAN_AUTH_ENABLE:-false}
+TORAN_AUTH_USER=${TORAN_AUTH_USER:-false}
+TORAN_AUTH_PASSWORD=${TORAN_AUTH_PASSWORD:-false}
```
Comment (cedvan) [inline scripts/config.sh:29]: Please add verification format to new params

File: scripts/config.sh [hunk 3]
```diff
@@ -23,6 +23,9 @@ TORAN_TOKEN_GITHUB=${TORAN_TOKEN_GITHUB:-false}
 TORAN_TRACK_DOWNLOADS=${TORAN_TRACK_DOWNLOADS:-false}
 TORAN_MONO_REPO=${TORAN_MONO_REPO:-false}
 TORAN_SECRET=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)
+TORAN_AUTH_ENABLE=${TORAN_AUTH_ENABLE:-false}
+TORAN_AUTH_USER=${TORAN_AUTH_USER:-false}
```
Comment (cedvan) [inline scripts/config.sh]: default must be : `toran`

File: scripts/config.sh [hunk 4]
```diff
@@ -23,6 +23,9 @@ TORAN_TOKEN_GITHUB=${TORAN_TOKEN_GITHUB:-false}
...
+TORAN_AUTH_PASSWORD=${TORAN_AUTH_PASSWORD:-false}
```
Comment (cedvan) [inline scripts/config.sh]: default must be : `toran`

File: scripts/install/nginx.sh [hunk 5]
```diff
@@ -1,5 +1,40 @@
 #!/bin/bash
 
+# Basic Auth
+echo "Detecting HTTP Basic Authentication Configuration"
+if [ "${TORAN_AUTH_ENABLE}" != "false" ]; then
+
+    if [ "${TORAN_AUTH_USER}" == "false" ]; then
```
Comment (cedvan) [inline scripts/install/nginx.sh]: check useless with default `toran` and format verification in `config.sh`

File: scripts/install/nginx.sh [hunk 6]
```diff
@@ -1,5 +1,40 @@
...
+        echo "ERROR: "
+        echo "  No TORAN_AUTH_USER provided."
+        exit 1
+    fi
+
+    if [ "${TORAN_AUTH_PASSWORD}" == "false" ]; then
```
Comment (cedvan) [inline scripts/install/nginx.sh]: check useless with default `toran` and format verification in `config.sh`

File: README.md [hunk 7]
```diff
@@ -106,6 +106,9 @@ Below is the complete list of available options that can be used to customize yo
...
+- **TORAN_AUTH_USER**: Configure the HTTP Basic Authentication Username
```
Comment (cedvan) [inline README.md]: add : Default to `toran`

File: README.md [hunk 8]
```diff
@@ -106,6 +106,9 @@ Below is the complete list of available options that can be used to customize yo
...
+- **TORAN_AUTH_PASSWORD**: Configure the HTTP Basic Authentication Password
```
Comment (cedvan) [inline README.md]: add : Default to `toran`

**Review by cedvan — CHANGES_REQUESTED** [review]

File: scripts/config.sh [hunk 9]
```diff
@@ -90,3 +93,15 @@ if [[ ! "${TORAN_CRON_TIMER_DAILY_TIME}" =~ ^[0-9]{2}:[0-9]{2}$ ]]; then
     echo "  Variable TORAN_CRON_TIMER_DAILY_TIME isn't valid ! (Format accepted : HH:MM)"
     exit 1
 fi
+
```
Comment (cedvan) [inline scripts/config.sh:96]: missing : 

```
# Checking Toran auth enable
if [ "${TORAN_AUTH_ENABLE}" != "true" ] && [ "${TORAN_AUTH_ENABLE}" != "false" ]; then
    echo "ERROR: "
    echo "  Variable TORAN_AUTH_ENABLE isn't valid ! (Values accepted : true/false)"
    exit 1
fi
```

**Review by cedvan — CHANGES_REQUESTED** [review]

**Review by cedvan — APPROVED** [review]

**Comment by cedvan** [comment]
Can you rebase to one commit please ?

**Comment by cedvan** [comment]
And just add small description with list changes, and i'll merge

**Comment by adamgoose** [comment]
I did put something in the changelog. Was there anything else/more specific you're looking for?

**Comment by cedvan** [comment]
Same changelog is ok :wink:

**Comment by cedvan** [comment]
No previous changelog was good, rollback changelog and just copy previous text from changelog in merge request description :smile:

**Comment by cedvan** [comment]
Thank for contributing :+1:

**Comment by adamgoose** [comment]
Thanks for merging! :) Is this going to build and tag as the latest 1.5.3 on Docker Hub?

**Pull request merged**